Show schematic description while hovering the mouser pointer over the file name

andresmmera opened the pull request:
The aim of this PR is to show a schematic description when the user hovers the mouse pointer over the project tree. In this sense, it was written a function for reading the schematic title from its frame.

Please refer to #763 for context
![screenshot from 2018-01-01 09-41-57](https://user-images.githubusercontent.com/13180689/34466654-7eac3866-eedb-11e7-959b-6737ebff0dc9.png)

guitorri commented:
Looks good to me. But IMHO there should be an option on the settings to enable/disable the tooltip.

andresmmera commented:
Ok, I've put that option under the "Application Settings" menu since I thought it was the most appropriate place. Additionally, I left the checkbox unchecked by default

![showdescriptionmenu](https://user-images.githubusercontent.com/13180689/36061941-a482374c-0e62-11e8-8d5b-fd8a97e7a334.png)

in3otd commented:
why there is a [commit](https://github.com/Qucs/qucs/pull/764/commits/3277589d8986874ec44bf9fca11b8f41e581e50a) for the lock function here? Also, it seems that a schematic is locked by default (?).

If a multiline description is used the tooltip show all the text on a single line, with `\n` separators; maybe this [depends](http://doc.qt.io/archives/qt-4.8/qtooltip.html#details) on the text format used.

The `QucsSettings.ShowDescriptionProjectTree` setting value is not saved on exit.

andresmmera commented:
> why there is a commit for the lock function here?

That shouldn't be there. I've just removed that from the commit history

> The QucsSettings.ShowDescriptionProjectTree setting value is not saved on exit.

Fixed here https://github.com/Qucs/qucs/pull/764/commits/5e7f9601f803be956cbd6129493cf9e9a8439f92

In addition to this, I've also added a [commit](https://github.com/Qucs/qucs/pull/764/commits/ccb1ef2a1024c90d91e806f0d276236993fee12b) to prevent those files where the description field is empty from showing the default text ("Title") in the tooltip

> If a multiline description is used the tooltip show all the text on a single line,

I've been doing some tests here... If I replace \n by the html breakline tag, < br > it works as duly

![screenshot from 2018-04-07 10-09-14](https://user-images.githubusercontent.com/13180689/38452918-a4c549ee-3a4d-11e8-8fdb-e5524d0f7b6d.png)

It may sound stupid, but I'm having problems to replace '\n' by "< br > at ProjectView::ReadDescription(QString). I think the proper way to do that should be:

`description.replace(QRegExp("[\\n]"),"<` br >"); [1]
or even `description.replace('\n',"<` br >");

but it doesn't work... Any guess?

P.S.: I'm adding a blank space after and before br because otherwhise the github wiki markdown replaces that by a breakline...

[1] http://www.qtcentre.org/threads/22060-removing-carriage-return-from-QString

in3otd commented on qucs/qucs/projectView.cpp:
```diff
@@ -157,3 +163,28 @@ ProjectView::exportSchematic()
+
+    description.replace("\\n", "<br>");
+    QucsDocument.close();
+    if (description == "Title") description = "";//Prevent schematics without description from showing "Title" in the tooltip
```
the description string is internationalized: in [schematic.cpp](https://github.com/Qucs/qucs/blob/develop/qucs/qucs/schematic.cpp#L98) the default value of `Frame_Text0` is `tr("Title")` so it would have been better to compare to `tr("Title")`.
Even better will be to show the description only if `showFrame` is set; the rationale is that the description is shown in the schematic and so has probably set to something sensible only if that is set..
This will also handle correctly the case when you open a schematic created with another Qucs language setting. E.g. when Qucs is set to use Italian the `Frame_Text0` is correctly set to *Titolo* but if you now open this schematic from a Qucs with language setting set to English it will check if the description is set to *Title* so the comparison will fail and it will show the *Titolo* string as description in the tooltip.

andresmmera replied:
I agree. 
Since this PR is already merged, I've opened a new one (#868) to fix this.
